File: backend/run_tests.py
#!/usr/bin/env python3
"""
Test runner script for the Travel Documentation Assistant API
"""
import os
import subprocess
import sys


def run_tests():
    """Run all tests with pytest"""
    print("🧪 Running tests for Travel Documentation Assistant API...")
    print("=" * 60)
    
    # Change to the backend directory
    os.chdir(os.path.dirname(os.path.abspath(__file__)))
    
    # Run pytest with verbose output
    cmd = [
        sys.executable, "-m", "pytest", 
        "test_main.py", 
        "-v", 
        "--tb=short",
        "--color=yes"
    ]
    
    try:
        result = subprocess.run(cmd, check=True)
        print("\n✅ All tests passed!")
        return 0
    except subprocess.CalledProcessError as e:
        print(f"\n❌ Tests failed with exit code {e.returncode}")
        return e.returncode
    except FileNotFoundError:
        print("❌ pytest not found. Please install test dependencies:")
        print("   pip install pytest pytest-asyncio httpx")
        return 1

if __name__ == "__main__":
    exit_code = run_tests()
    sys.exit(exit_code)

File: frontend/run_tests.js
#!/usr/bin/env node
/**
 * Test runner script for the Travel Documentation Assistant Frontend
 */
const { spawn } = require('child_process');
const path = require('path');

function runTests() {
  console.log('🧪 Running tests for Travel Documentation Assistant Frontend...');
  console.log('=' * 60);
  
  const testCommand = process.argv[2] || 'test';
  const args = process.argv.slice(3);
  
  let command, commandArgs;
  
  switch (testCommand) {
    case 'watch':
      command = 'npm';
      commandArgs = ['run', 'test:watch'];
      break;
    case 'coverage':
      command = 'npm';
      commandArgs = ['run', 'test:coverage'];
      break;
    case 'ci':
      command = 'npm';
      commandArgs = ['run', 'test', '--', '--ci', '--coverage', '--watchAll=false'];
      break;
    default:
      command = 'npm';
      commandArgs = ['run', 'test'];
  }
  
  // Add any additional args
  commandArgs.push(...args);
  
  const child = spawn(command, commandArgs, {
    stdio: 'inherit',
    shell: true,
    cwd: path.resolve(__dirname)
  });
  
  child.on('close', (code) => {
    if (code === 0) {
      console.log('\n✅ All tests passed!');
    } else {
      console.log(`\n❌ Tests failed with exit code ${code}`);
    }
    process.exit(code);
  });
  
  child.on('error', (error) => {
    console.error('❌ Failed to start test process:', error);
    process.exit(1);
  });
}

if (require.main === module) {
  runTests();
}

module.exports = { runTests };
